src/iwant_bot/pool.py

```diff
@@ -43,33 +47,26 @@ def update_requests_from_storage(self):
                 destination.append(req)
 
     def make_pairs(self, activity):
+        ignore_list = self.ignore_list
         source = self.req_by_activities[activity]
         destination = self.pairs[activity]
-        while len(source) > 1:
-            request1 = source.pop()
-            request2 = source.pop()
-            paired = self.pair(request1, request2)
-            destination.append(paired)
-
-    async def check_expired(self):
-        pass
-
-    async def coro(self):
-        pass
+        if len(source) > 1:
+            for request1 in source:
+                source.remove(request1)
+                for request2 in source:
+                    if ignore_list.mutual_ignore_check(request1.person_id, request2.person_id):
+                        continue
+                    paired = self.pair(request1, request2)
+                    source.remove(request2)
+                    destination.append(paired)
+                    break
+                source.append(request1)
```

Comment:
Transform this if/for/for piece to linear code using `zip` `filter` and make uniqueness by overriding `__eq__`
```
requests = set(requests)
pairs = map(make_pair, zip(requests))
pairs = filter(exclude_self, pairs)
pairs = filter(exclude_ignored, pairs)
...
remaining_requests = requests - flatten(pairs)
```